SimMuon/GEMDigitizer/src/ME0SimpleModel.cc

```diff
@@ -0,0 +1,303 @@
+  bx = static_cast<int> (std::round((timeDifference) / bxwidth_));
+
+  // check time
+  const bool debug(false);
+  if (debug)
+  {
+    std::cout << "checktime " << "bx = " << bx << "\tdeltaT = " << timeDifference << "\tsimT =  " << simhitTime
+        << "\trefT =  " << referenceTime << "\ttof = " << tof << "\tavePropT =  " << averagePropagationTime
+        << "\taveRefPropT = " << halfStripLength / signalPropagationSpeedTrue << std::endl;
+  }
+  return bx;
+}
+
+void ME0SimpleModel::simulateNoise(const ME0EtaPartition* roll, CLHEP::HepRandomEngine* engine)
+{
+  if (!doBkgNoise_)
+    return;
+  const ME0DetId me0Id(roll->id());
+  const int nstrips(roll->nstrips());
+  double trArea(0.0);
+  double trStripArea(0.0);
+
+  if (me0Id.region() == 0)
+  {
+    throw cms::Exception("Geometry")
+        << "ME0Synchronizer::simulateNoise() - this ME0 id is from barrel, which cannot happen.";
+  }
+  const TrapezoidalStripTopology* top_(dynamic_cast<const TrapezoidalStripTopology*>(&(roll->topology())));
+  const float striplength(top_->stripLength());
+  trStripArea = (roll->pitch()) * striplength;
+  trArea = trStripArea * nstrips;
+  const int nBxing(maxBunch_ - minBunch_ + 1);
+  const float rollRadius(fixedRollRadius_ ? top_->radius() : 
+			 top_->radius() + CLHEP::RandFlat::shoot(engine, -1.*top_->stripLength()/2., top_->stripLength()/2.));
+
+  const float rSqrtR = rollRadius * sqrt(rollRadius);
+
+  //calculate noise from model
+  double averageNeutralNoiseRatePerRoll = 0.;
+  double averageNoiseElectronRatePerRoll = 0.;
+  double averageNoiseRatePerRoll = 0.;
+  if (me0Id.station() != 1)
+  {
+    throw cms::Exception("Geometry") << "ME0SimpleModel::simulateNoise() - this ME0 id is from station 1, which cannot happen: " <<roll->id() << "\n";
+  }
+  else
+  {
+    averageNeutralNoiseRatePerRoll = ME0NeuBkgParam0 * rollRadius* TMath::Exp(ME0NeuBkgParam1/rSqrtR) + ME0NeuBkgParam2/rSqrtR + ME0NeuBkgParam3/(sqrt(rollRadius));
+
+    //simulate electron background for ME0
+    if (simulateElectronBkg_)
+      averageNoiseElectronRatePerRoll = ME0ElecBkgParam0 * rSqrtR* TMath::Exp(ME0ElecBkgParam1/rSqrtR) + ME0ElecBkgParam2/rSqrtR + ME0ElecBkgParam3/(sqrt(rollRadius));
+
+    averageNoiseRatePerRoll = averageNeutralNoiseRatePerRoll + averageNoiseElectronRatePerRoll;
+    averageNoiseRatePerRoll *= instLumi_*rateFact_*1.0/referenceInstLumi_;  
+  }
+
+  //simulate intrinsic noise
+  if(simulateIntrinsicNoise_)
+  {
+    const double aveIntrinsicNoisePerStrip(averageNoiseRate_ * nBxing * bxwidth_ * trStripArea * 1.0e-9);
+    for(int j = 0; j < nstrips; ++j)
+    {
+      CLHEP::RandPoissonQ randPoissonQ(*engine, aveIntrinsicNoisePerStrip);
+      const int n_intrHits(randPoissonQ.fire());
+    
+      for (int k = 0; k < n_intrHits; k++ )
+      {
+        const int time_hit(static_cast<int>(CLHEP::RandFlat::shoot(engine, nBxing)) + minBunch_);
+        std::pair<int, int> digi(k+1,time_hit);
+        strips_.insert(digi);
```

Comment:
just use `strips_.emplace(k+1,time_hit)` to avoid the copy (also several instances further down)